controllers/mongodb_users.go

```diff
@@ -5,18 +5,26 @@ import (
 
 	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	"go.uber.org/zap"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 // ensureUserResources will check that the configured user password secrets can be found
 // and will start monitor them so that the reconcile process is triggered every time these secrets are updated
-func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity) error {
+func (r ReplicaSetReconciler) ensureUserResources(mdb mdbv1.MongoDBCommunity, log *zap.SugaredLogger) error {
 	for _, user := range mdb.GetScramUsers() {
 		secretNamespacedName := types.NamespacedName{Name: user.PasswordSecretName, Namespace: mdb.Namespace}
 		if _, err := secret.ReadKey(r.client, user.PasswordSecretKey, secretNamespacedName); err != nil {
 			if apiErrors.IsNotFound(err) {
-				return fmt.Errorf(`user password secret "%s" not found: %s`, secretNamespacedName, err)
+				// check for SCRAM secret as well
+				scramSecretName := mdb.GetAgentScramCredentialsNamespacedName()
```

Comment:
this should be `user.GetScramCredentialsSecretName()` (there is a different set of creds for the Agent)

Reply:
there seems to be leftover `GetAgentScramCredentialsNamespacedName` which is not used anymore and we have just one credentials -- so I will remove the former.